{
  "WorkItem": {
    "AffectedComponent": {
      "Name": "",
      "DisplayName": ""
    },
    "ClosedComment": "",
    "ClosedDate": "2013-07-19T10:27:16.93-07:00",
    "CommentCount": 2,
    "Custom": null,
    "Description": "The following gives an error on the 5th line of the repro below, saying:\r\n\r\nExported variable 'instance' has or is using private type 'Foo'.\r\n\r\nThis seems to be an incorrect error message.\r\n\r\n```js\r\nclass Foo {\r\n\tpublic get foo() {\r\n\t\tvar instance: Foo;\r\n\t\t\r\n\t\treturn instance;\r\n\t}\r\n}\r\n\r\nexport var x = function() {\r\n\t\r\n}\r\n```",
    "LastUpdatedDate": "2014-01-10T10:34:33.287-08:00",
    "PlannedForRelease": "",
    "ReleaseVisibleToPublic": false,
    "Priority": {
      "Name": "Unassigned",
      "Severity": 0,
      "Id": 0
    },
    "ProjectName": "typescript",
    "ReportedDate": "2013-06-18T17:49:52.397-07:00",
    "Status": {
      "Name": "Closed",
      "Id": 4
    },
    "ReasonClosed": {
      "Name": "Fixed"
    },
    "Summary": "Incorrect error message in class get member that returns an instance when also exporting something",
    "Type": {
      "Name": "Issue",
      "Id": 3
    },
    "VoteCount": 1,
    "Id": 1166
  },
  "FileAttachments": [],
  "Comments": [
    {
      "Message": "Thanks for reporting.  An interesting bug.  We'll take a look!",
      "PostedDate": "2013-06-19T00:24:45.69-07:00",
      "Id": 103379
    },
    {
      "Message": "This is fixed as of check-in http://typescript.codeplex.com/SourceControl/changeset/f4fbbb68674791e03a8702b49ea012ecad29c002 .  We'll add a specific regression test for this also.",
      "PostedDate": "2013-07-19T10:27:12.423-07:00",
      "Id": 106644
    }
  ]
}